web/client/plugins/Map.jsx

```diff
@@ -112,29 +114,49 @@ const {handleCreationLayerError, handleCreationBackgroundError, resetMapOnInit}
  *  - name is a unique name for the tool
  *  - impl is a placeholder (“{context.ToolName}”) where ToolName is the name you gave the tool in plugins.js (TestSupportLeaflet in our example)
  *
+ * You can also specify a list if fonts that will be loaded before the **openlayers** map gets rendered (on willMount lyfe)
+ * For each font you can specify an object describing the variation. The object can contain weight, style, and stretch properties.
+ * If a property is not present it will default to normal.
+ * ```
+ * {
+ *    "name": "Map",
+ *    "cfg": {
+ *      "fonts": {
+ *        'FontAwesome': { weight: 400 },
+ *        'AnotherFontFamily': {}
+ *      }
+ *    }
+ *  }
+ * ```
+ * For more info on metadata visit [fontfaceobserver](https://github.com/bramstein/fontfaceobserver)
+ *
  * @memberof plugins
  * @class Map
  * @prop {array} additionalLayers static layers available in addition to those loaded from the configuration
  * @static
  * @example
-  * // Adding a layer to be used as a source for the elevation (shown in the MousePosition plugin configured with showElevation = true)
-  * {
-  *   "cfg": {
-  *     "additionalLayers": [{
-  *         "type": "wms",
-  *         "url": "http://localhost:8090/geoserver/wms",
-  *         "visibility": true,
-  *         "title": "Elevation",
-  *         "name": "topp:elevation",
-  *         "format": "application/bil16",
-  *         "useForElevation": true,
-  *         "nodata": -9999,
-  *         "hidden": true
-  *      }]
-  *   }
-  * }
+ * // Adding a layer to be used as a source for the elevation (shown in the MousePosition plugin configured with showElevation = true)
+ * {
+ *   "cfg": {
+ *     "additionalLayers": [{
+ *         "type": "wms",
+ *         "url": "http://localhost:8090/geoserver/wms",
+ *         "visibility": true,
+ *         "title": "Elevation",
+ *         "name": "topp:elevation",
+ *         "format": "application/bil16",
+ *         "useForElevation": true,
+ *         "nodata": -9999,
+ *         "hidden": true
+ *      }]
+ *   }
+ * }
  *
  */
+
+const updateCanRenderState = (canRender) => () => ({
```

Comment:
???

Reply:
This function updates the internal state, [setState accepts also functions](https://medium.freecodecamp.org/functional-setstate-is-the-future-of-react-374f30401b6b).
since i use it twice, i've created a function, i'll add a description above.